app/containers/App/reducer.js

```diff
@@ -37,12 +37,12 @@ function appReducer(state = initialState, action) {
         .setIn(['userData', 'repositories'], false);
     case LOAD_REPOS_SUCCESS:
       return state
-        .setIn(['userData', 'repositories'], action.repos)
+        .setIn(['userData', 'repositories'], fromJS(action.repos))
         .set('loading', false)
         .set('currentUser', action.username);
```

Comment:
If we're really trying to eek out the most performance possible then multiple `set`/`setIn`s could be wrapped in `withMutations` to avoid creating a new copy of the underlying Map for each operation ([read more](https://facebook.github.io/immutable-js/docs/#/Map/withMutations)).

This could instead be considered as:
```js
return state.withMutations(state => state
  .setIn(['userData', 'repositories'], fromJS(action.repos))
  .set('loading', false)
  .set('currentUser', action.username)
);
```

I'm glad someone else picked up on that toJS HOC, thanks for bringing it up!